### PR TITLE
#567: First step in migration to Swift 3.x in AtlasSDK

### PR DESCRIPTION
**Pull request by bonkey**
#567: First step in migration to Swift 3.x in AtlasSDK

### CODE REVIEW

**Comment by bonkey** [comment]
@hani-ibrahim Far from finished, but I want to make it gentle

**Comment by bonkey** [comment]
Will make few others PR from-to `issue-29`, so you can merge after you review

File: AtlasSDK/APIRequest.swift [hunk 1]
```diff
@@ -7,33 +7,36 @@ import Foundation
...
         completions = []
     }
 
-    public mutating func execute(completion: AtlasAPIResult<T> -> Void) {
-        completions.append(completion)
+    public mutating func execute(_ completion: @escaping (AtlasAPIResult<T>) -> Void) {
+        self.completions.append(completion)
+
+        let completions = self.completions.reversed()
+        let successHandler = self.successHandler
```
Comment (hani-ibrahim) [inline AtlasSDK/APIRequest.swift]: why?

File: AtlasSDK/APIRequest.swift [hunk 2]
```diff
@@ -7,33 +7,36 @@ import Foundation
...
+                DispatchQueue.main.async {
+                    completions.forEach { $0(.failure(error, self)) }
                 }
 
             case .success(let response):
-                if let parsedResponse = self.successHandler(response) {
-                    dispatch_async(dispatch_get_main_queue()) {
-                        self.completions.reverse().forEach { $0(.success(parsedResponse)) }
+                if let parsedResponse = successHandler(response) {
+                    DispatchQueue.main.async {
```
Comment (hani-ibrahim) [inline AtlasSDK/APIRequest.swift]: Wow, seams like we can drop the `Async` struct in the UI

File: AtlasSDK/AtlasSDK/Utils/Networking/URLStringConvertible.swift [hunk 3]
```diff
@@ -18,22 +18,18 @@ extension String: URLStringConvertible {
 
 }
 
-extension NSURL: URLStringConvertible {
+extension URL: URLStringConvertible {
 
     public var URLString: String {
```
Comment (hani-ibrahim) [inline AtlasSDK/AtlasSDK/Utils/Networking/URLStringConvertible.swift]: is it needed anymore?

File: AtlasSDK/AtlasSDK/Utils/SwiftyJSON/SwiftyJSON.swift [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 //  SwiftyJSON.swift
 //
-//  Copyright (c) 2014 Ruoyu Fu, Pinglin Tang
+//  Copyright (c) 2014 - 2016 Ruoyu Fu, Pinglin Tang
```
Comment (hani-ibrahim) [inline AtlasSDK/AtlasSDK/Utils/SwiftyJSON/SwiftyJSON.swift]: 2014 ! 😐

File: AtlasUI/AtlasUI/CheckoutSummaryLayout.swift [hunk 5]
```diff
@@ -1,4 +1,3 @@
-// swiftlint:disable line_length
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/CheckoutSummaryLayout.swift]: why? 😖 
I want this file to be consistent to compare different types easily and not spited over many lines

**Comment by hani-ibrahim** [comment]
👍

File: AtlasSDK/APIRequest.swift (hunk 2 above)
Comment (bonkey) [inline AtlasSDK/APIRequest.swift]: 👍

File: AtlasSDK/APIRequest.swift (hunk 1 above)
Comment (bonkey) [inline AtlasSDK/APIRequest.swift]: Yeah, don't know the reasoning yet, but compiler tells:

```
error: closure cannot implicitly capture a mutating self parameter
self.completions.reversed().forEach { $0(.failure(error, self)) }
^
```

File: AtlasSDK/AtlasSDK/Utils/Networking/URLStringConvertible.swift (hunk 3 above)
Comment (bonkey) [inline AtlasSDK/AtlasSDK/Utils/Networking/URLStringConvertible.swift]: Probably yes

File: AtlasUI/AtlasUI/CheckoutSummaryLayout.swift (hunk 5 above)
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryLayout.swift]: Have no idea, maybe after merge? Nothing on purpose, but it seems like [broken window](https://en.wikipedia.org/wiki/Broken_windows_theory)

**Pull request merged**